src/compiler/interface.jl

```diff
@@ -65,11 +63,13 @@ Base.adjoint(f::Function) = x -> gradient(f, x)[1]
 
 # TODO store ids only
 struct Params
-  order::Buffer{Any, Vector{Any}}
+  order::Buffer # {Any, Vector{Any}}
   params::IdSet{Any}
-  Params() = new(Buffer([], false), IdSet())
 end
 
+Params() = Params(Buffer([], false), IdSet())
+Params(xs) = Params(Buffer(xs, false), IdSet(xs))
```

Comment:
```suggestion
Params(xs) = Params(Buffer(xs, false), IdSet(xs))
Params(ps::Params) = ps
```

Reply:
Better to type`xs::Vector`

Does this show up in practice?

Reply:
I know of only one place which is L240 of this file. It is not needed obviously, but it depends on what you want to fix. The previous `Params(xs) = push!(Params(), xs)` definition handled the case where `xs isa Params`. Without this constructor, we'd be changing behavior.

Reply:
Seems like an odd design choice - `map` should be careful not to do such things.

Reply:
We can handle the case either way, but that doesn't allow map to be bad.